internal/app/server/open_saves_test.go

```diff
@@ -99,6 +112,31 @@ func TestOpenSaves_HealthCheck(t *testing.T) {
 	if want := healthgrpc.HealthCheckResponse_SERVING; got.Status != want {
 		t.Errorf("hc.Check got: %v, want: %v", got.Status, want)
 	}
+	cancel()
```

Comment:
Why do we need to cancel the context? Do we want to test the behavior of the server after `cancel()`?

Reply:
I added a comment here to address this. I also added tests after this to test NOT_SERVING.

Reply:
lgtm